notebooks/source/bayesian_hierarchical_linear_regression.ipynb

```diff
@@ -0,0 +1,677 @@
+    "\n",
+    "We were are asked to predict every patient's FVC measurement for every possible week in the [-12, 133] interval, and the confidence for each prediction. In other words: we were asked fill a matrix like the one below, and provide a confidence score for each prediction:\n",
+    "\n",
+    "<img src=\"https://i.ibb.co/0Z9kW8H/matrix-completion.jpg\" alt=\"drawing\" width=\"600\"/>\n",
+    "\n",
+    "The task was perfect to apply Bayesian inference. However, the vast majority of solutions shared by Kaggle community used discriminative machine learning models, disconsidering the fact that most discriminative methods are very poor at providing realistic uncertainty estimates. Because they are typically trained in a manner that optimizes the parameters to minimize some loss criterion (e.g. the predictive error), they do not, in general, encode any uncertainty in either their parameters or the subsequent predictions. Though many methods can produce uncertainty estimates either as a by-product or from a post-processing step, these are typically heuristic based, rather than stemming naturally from a statistically principled estimate of the target uncertainty distribution [2].\n",
+    "\n",
+    "## 2. Modelling: Bayesian Hierarchical Linear Regression with Partial Pooling\n",
+    "The simplest possible linear regression, not hierarchical, would assume all FVC decline curves have the same $\\alpha$ and $\\beta$. That's the **pooled model**. In the other extreme, we could assume a model where each patient has a personalized FVC decline curve, and **these curves are completely unrelated**. That's the **unpooled model**, where each patient has completely separate regressions.\n",
+    "\n",
+    "Here, I'll use the middle ground: **Partial pooling**. Specifically, I'll assume that while $\\alpha$'s and $\\beta$'s are different for each patient as in the unpooled case, **the coefficients all share similarity**. We can model this by assuming that each individual coefficient comes from a common group distribution. The image below represents this model graphically:\n",
+    "\n",
+    "<img src=\"https://i.ibb.co/H7NgBfR/Artboard-2-2x-100.jpg\" alt=\"drawing\" width=\"600\"/>\n",
+    "\n",
+    "Mathematically, the model is described by the following equations:\n",
+    "$$\n",
+    "\\mu_{\\alpha} \\sim \\mathcal{N}(0, 100) \\\\\n",
+    "\\sigma_{\\alpha} \\sim |\\mathcal{N}(0, 100)| \\\\\n",
+    "\\mu_{\\beta} \\sim \\mathcal{N}(0, 100) \\\\\n",
+    "\\sigma_{\\beta} \\sim |\\mathcal{N}(0, 100)| \\\\\n",
+    "\\alpha_i \\sim \\mathcal{N}(\\mu_{\\alpha}, \\sigma_{\\alpha}) \\\\\n",
+    "\\beta_i \\sim \\mathcal{N}(\\mu_{\\beta}, \\sigma_{\\beta}) \\\\\n",
+    "\\sigma \\sim \\mathcal{N}(0, 100) \\\\\n",
+    "FVC_{ij} \\sim \\mathcal{N}(\\alpha_i + t \\beta_i, \\sigma)\n",
+    "$$\n",
```

Comment:
Inside `$$` pairs, you can use `\end{align} ... \end{align}` to align those formulas line-by-line (see the `time_series_forecasting.ipynb` notebook for an example).